172.md

```diff
@@ -0,0 +1,100 @@
+NIP-172
+=======
+
+Moderated Communities (Reddit Style)
+------------------------------------
+
+`draft` `optional` `author:vitorpamplona` `author:arthurfranca`
+
+The goal of this NIP is to create moderator-approved public communities around a topic. It defines the replaceable event `kind:34550` to define the community and the current list of moderators/administrators. Users that want to post into the community, simply tag any Nostr event with the community's `a` tag (See [NIP-33](33.md)). Moderators issue an approval event `kind:4550` that links the community with the new post.
+
+# Community Definition
+
+`Kind:34550` SHOULD include any field that helps define the community and the set of moderators.
+
+```js
+{
+  "id": "<32-bytes lowercase hex-encoded SHA-256 of the the serialized event data>",
+  "pubkey": "<32-bytes lowercase hex-encoded public key of the event creator>",
+  "created_at": "<Unix timestamp in seconds>",
+  "kind": 34550,
+  "tags": [
+    ["d", "<Community name>"],
+    ["description", "<Community description>"],
+    ["image", "<Community image url>", "<Width>x<Height>"],
+	
+    //.. other tags relevant to defining the community
+
+    // moderators
+    ["p", "<32-bytes hex of a pubkey1>", "<optional recommended relay URL>", "moderator"],
+    ["p", "<32-bytes hex of a pubkey2>", "<optional recommended relay URL>", "moderator"],
+    ["p", "<32-bytes hex of a pubkey3>", "<optional recommended relay URL>", "moderator"],
+	
+    // relays used by the community 
+    ["relay", "<relay hosting author kind 0>", "author"],
+    ["relay", "<relay where to post requests to and fetch approvals from>"], 
+    ["relay", "<relay where to post requests to and fetch approvals from>"]
+  ]
+}
+```
+
+# New Post Request
+
+Any Nostr event can be a post request. Clients MUST add the community's `a` tag to the new post event in order to be presented for the moderator's approval.
+
+```js
+{
+  "id": "<32-bytes lowercase hex-encoded SHA-256 of the the serialized event data>",
+  "pubkey": "<32-bytes lowercase hex-encoded public key of the event creator>",
+  "created_at": "<Unix timestamp in seconds>",
+  "kind": 1,
+  "tags": [
+    ["a", "34550:<Community event author pubkey>:<d-identifier of the community>", "<Optional relay url>"],
```

Comment:
When modifying a community, the author of that event could be different from the author of the community creation event.  

So, this kind of 'a' tag may require us to hunt for a possibly deleted original 'community creation' event to fetch all  notes in a community?  

May be I am missing something here. Sorry if this question was silly.

Reply:
No, the author cannot be different. Replaceable events can only be replaced by the original author.